apps/avifenc.c

```diff
@@ -1561,6 +1576,11 @@ int main(int argc, char * argv[])
         goto cleanup;
     }
 
+    if (noOverwrite && file_exists(outputFilename)) {
+        fprintf(stderr, "ERROR: output file %s already exists and --no-overwrite was specified\n", outputFilename);
+        goto cleanup;
+    }
```

Comment:
Suppose the specified output file pathname is in a nonexistent or nonwritable directory. If the encoding takes a long time, we will also waste time encoding before realizing that the output file cannot be opened. This is another reason for opening the output file before encoding starts.

My personal preference is to go for simpler error handling unless errors are common. I haven't inspected avifenc.c carefully to know whether error handling is simpler if we open the output file before or after encoding starts.

Reply:
Both approaches have pros and cons... But even if errors should be rare, I would find deleting a user's file in case of error a bigger problem than wasting time in case we don't have write access. (This is the scenario where out.avif already exists, I run 'avifenc' outputting to out.avif, avifenc fails midway, and out.avif is either empty, or we would have deleted it in the cleanup).
Still I don't have a super strong opinion so I'm fine to go with whichever version you prefer or find easier.

Reply:
A common case where encoding exits part-way through is simply when the encoding job is cancelled or killed. Creating the output file before encoding, in this case, will litter the filesystem with empty files -- which is annoying to clean up.